pkg/providers/tinkerbell/assert.go

```diff
@@ -403,41 +444,64 @@ func ExtraHardwareAvailableAssertionForRollingUpgrade(catalogue *hardware.Catalo
 		// will account for the same selector being specified on different groups.
 		requirements := minimumHardwareRequirements{}
 
-		maxSurge := 1
-		if spec.Cluster.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy != nil {
-			maxSurge = spec.Cluster.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy.RollingUpdate.MaxSurge
+		if spec.Cluster.Spec.KubernetesVersion != *current.ClusterK8sVersion() || eksaVersionUpgrade {
+			if err := ensureCPHardwareAvailability(spec, current, requirements); err != nil {
+				return err
+			}
 		}
-		err := requirements.Add(
-			spec.ControlPlaneMachineConfig().Spec.HardwareSelector,
-			maxSurge,
-		)
-		if err != nil {
+
+		if err := ensureWorkerHardwareAvailability(spec, current, requirements, eksaVersionUpgrade); err != nil {
+			return err
+		}
+
+		if spec.HasExternalEtcd() {
+			return fmt.Errorf("external etcd upgrade is not supported")
+		}
```

Comment:
Wsa this check in this func before? If not I don't think its necessary, leave other validations to validate etcd is unspecified.

Reply:
Yes this case was already present in the function i had to just split up the function a bit as the complexity for over 10.!